code/Data.py

```diff
@@ -14,9 +20,17 @@ def __init__(self, src):
             for _, row in src:  # Else given rows so no processing just add
                 self.add(row)
 
-    def add(self, xs):
-        # Filler code this is TODO
-        self.cols = xs
+    
+    def add(self, xs: Row):
+
+        if not self.cols:
+            self.cols = Cols(xs) 
+        else:
+            row = xs if type(xs)==Row else Row(xs)    
+            self.rows.append(row)
+            for todo in [self.cols.x, self.cols.y]:
+                for col in todo:
+                    col.add(row["cells"][col.c])
```

Comment:
Adds data to col, although I might have used an incorrect variable for column position. Which I'll amend